### PR TITLE
Task2: Viachaslau Saroka

### PR DESCRIPTION
**Pull request by PTaHHHa**

### CODE REVIEW

File: Tasks/Task2/query09.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+SELECT Purchasing.Vendor.Name AS VendorName,
+       Product.Name AS ProductName
+FROM Purchasing.Vendor
+LEFT JOIN Purchasing.ProductVendor
+       ON Purchasing.Vendor.BusinessEntityID = Purchasing.ProductVendor.BusinessEntityID
+LEFT JOIN Production.Product
+       ON Production.Product.ProductID = Purchasing.ProductVendor.ProductID
+ORDER BY VendorName,
+       ProductName
+;
```
Comment (zeydlitz) [inline Tasks/Task2/query09.sql]: lgtm!

**Review by zeydlitz — COMMENTED** [review]
nice work

File: Tasks/Task2/query02.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+SELECT TOP 37 Person.BusinessEntityID,
```
Comment (YauheniD) [inline Tasks/Task2/query02.sql]: top 37 is not smth that was defined by the task

File: Tasks/Task2/query03.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
+SELECT Sales.Customer.CustomerID,
+       CASE 
+              WHEN Person.Person.MiddleName IS NULL
```
Comment (YauheniD) [inline Tasks/Task2/query03.sql]: there is an easier way to concat and check mid name for null

File: Tasks/Task2/query03.sql [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                                   ' ',
+                                   Person.Person.MiddleName,
+                                   ' ',
+                                   Person.Person.LastName
+                                   )
+              END AS CustomerFullName,
+       SalesOrderID,
+       OrderDate
+FROM Sales.SalesOrderHeader
+FULL OUTER JOIN Sales.Customer
```
Comment (YauheniD) [inline Tasks/Task2/query03.sql]: FULL OUTER is overkill here

File: Tasks/Task2/query04.sql [hunk 5]
```diff
@@ -0,0 +1,12 @@
+SELECT Person.Person.BusinessEntityID,
+       Person.Person.LastName,
+       Person.Person.FirstName
+FROM Person.Person
+LEFT JOIN Person.BusinessEntityAddress
+       ON Person.Person.BusinessEntityID = Person.BusinessEntityAddress.BusinessEntityID
+LEFT JOIN Person.Address
+       ON Person.BusinessEntityAddress.AddressID = Person.Address.AddressID
+WHERE Person.Address.AddressID IS NULL
+       OR Person.BusinessEntityAddress.AddressTypeID != 2
```
Comment (YauheniD) [inline Tasks/Task2/query04.sql]: what if that id would be reassigned to other value?

File: Tasks/Task2/query05.sql [hunk 6]
```diff
@@ -0,0 +1,19 @@
+SELECT HumanResources.Employee.BusinessEntityID,
+       Person.Person.LastName,
+       CASE 
+              WHEN Person.PersonPhone.PhoneNumberTypeID = 1
+                     THEN Person.PersonPhone.PhoneNumber
+              WHEN Person.PersonPhone.PhoneNumberTypeID != 1
```
Comment (YauheniD) [inline Tasks/Task2/query05.sql]: could be set else instead all this

File: Tasks/Task2/query07.sql [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+                     END AS 'Has Orders in 2013'
+       FROM Person.Person
+       LEFT JOIN Sales.Customer
+              ON Sales.Customer.PersonID = Person.Person.BusinessEntityID
+       LEFT JOIN Sales.SalesOrderHeader
+              ON Sales.SalesOrderHeader.CustomerID = Sales.Customer.CustomerID
+                     AND YEAR(OrderDate) = 2013
+       WHERE Person.Person.FirstName = 'Zoe'
+       ) AS a
+WHERE a.RowNumber = 1
```
Comment (YauheniD) [inline Tasks/Task2/query07.sql]: a bit confused what row num for

**Pull request closed without merging**